### ISSUE TITLE
[GUI] Suggest a better sub assets name "xxx xxx xxx" instead of "xxx/xxx/xxx"

### ISSUE THREAD

**Issue by Kin20101** [RavenProject/Ravencoin#433]
I think "xxx xxx xxx" is better than "xxx/xxx/xxx" in wallet. If there is a choice to select how to show the name, cool.

![111](https://user-images.githubusercontent.com/25874991/48166011-4ae95900-e322-11e8-893a-9601dfb588a5.jpg)
![222](https://user-images.githubusercontent.com/25874991/48166014-4ae95900-e322-11e8-9822-215533116c27.jpg)
![333](https://user-images.githubusercontent.com/25874991/48166015-4b81ef80-e322-11e8-87c5-be8a0dbc62a7.jpg)

**Comment by russkidooski** [comment]
kind of like a drop down menu per mainasset, i like it.

**Comment by MSFTserver** [comment]
i mean this is just a database its up to viewing program how to list stuff, im all for this way of listing tgem on the explorers would much easier searching RAVEN.OG and having a latter under in then RAVEN.OG/ being repeated because it was the parent

**Comment by AdamBLevine** [comment]
Apologies if this is a dumb comment, i'm just starting to dig in.

For the protocol, you want to have something that is very difficult to make mistakes with, but you want metadata embedded at creation or subsequently which a wallet can use to get the user friendly names.  

So in Counterparty we have ASSET.SUBASSET with an approved subset of characters which can be used in a valid name.  We then have "Blockchain Validated Asset Metadata"  (something like this) http://xcp.coindaddy.io/CZAR.json which is hosted offchain but has its hash anchored in the blockchain and so is tamper resistant.

How much have you drawn from the problems solved by the Counterparty project?  We covered all this ground over the course of four years and definitely have learnings that could be applied if you weren't paying attention.

**Comment by wolfsokta** [comment]
Asset explorers are free to display the sub-assets how they see fit. QT could also display them it a tree like structure but we decided to keep it simple for now.

**Comment by TronBlack** [comment]
Hi Adam.   Thanks for digging in on this.  We've used IPFS for the meta-data which solves two problems.  First, it solves the persistence issue - assuming the files are pinned by someone.   I've worked with Counterparty and Open Assets. Both suffered from missing meta-data when the JSON data referenced by the shortened URLs disappear.    By storing the IPFS hash on-chain as the file reference, which is also the hash of the data, it automatically makes it tamper-resistant, perhaps to the point of being tamper-proof.
